voluptuous/validators.py

```diff
@@ -794,3 +793,57 @@ def __call__(self, v):
 
     def __repr__(self):
         return 'Unordered([{}])'.format(", ".join(repr(v) for v in self.validators))
+
+
+
+class Number(object):
+    """
+    Verify the maximum number of digits that are present in the number(Precision),
+    and the decimal places(Scale)
+
+    :raises Invalid: If the value does not match the provided Precision and Scale.
+
+    >>> schema = Schema({"number" : Number(precision=6, scale=2)})
+    >>> out_ = schema({"number": '1234.01'})
+    """
+
+    def __init__(self, precision=None, scale=None, msg=None):
+        self.precision = precision
+        self.scale = scale
+        self.msg = msg
+
+    def __call__(self, v):
+        """
+        :param v: is a number enclosed with string
+        :return: Decimal number
+        """
+        precision, scale, number = self.__get_precision_scale(v)
+
+        if self.precision is not None and self.scale is not None and\
+            precision != self.precision and scale != self.scale:
+            raise Invalid(self.msg or "Precision must be equal to %s, and Scale must be equal to %s" %(self.precision, self.scale))
+        else:
+            if self.precision is not None and precision != self.precision:
+                raise Invalid(self.msg or "Precision must be equal to %s"%self.precision)
+
+            if self.scale is not None and scale != self.scale :
+                raise Invalid(self.msg or "Scale must be equal to %s"%self.scale)
+
+        return number
+
+    def __repr__(self):
+        return ('Number(precision=%s, scale=%s, msg=%s)' % (self.precision, self.scale, self.msg))
+
+    def __get_precision_scale(self, number):
```

Comment:
Let's make `__get_precision_scale` internal instead of `_get_precision_scale`. Refer http://stackoverflow.com/a/6930223/1799475